cmake/config/default.cmake

```diff
@@ -1,42 +1,42 @@
-add_config(
-    CONFIG_NAME CMAKE_BUILD_TYPE
-    CONFIG_TYPE STRING
-    DEFAULT_VAL Release
-    DESCRIPTION "The target build type"
-    OPTIONS Release Debug HelixQAC
-)
-
-add_config(
-    CONFIG_NAME TARGET_ARCH
+hypervisor_add_config(
+    CONFIG_NAME HYPERVISOR_TARGET_ARCH
```

Comment:
Just like with CMake functions/macros, FetchContent behaves such that subprojects can trample over cache variables and build configurations. For example, PAL has its own TARGET_ARCH configuration that works a little differently than the hypervisor project. Again, the solution is to rename all cache variables and build configs so that they are globally unique.